app/src/main/java/com/eslamwaheed/mvvmdemo/viewmodels/UnsplashViewModel.kt

```diff
@@ -16,11 +16,33 @@ class UnsplashViewModel @Inject constructor(private val repository: UnsplashRepo
 
     val unsplashSearchResponseLiveData = MutableLiveData<UnsplashSearchResponse>()
 
+    val photosLiveData = MutableLiveData<ArrayList<UnsplashSearchResponse.Result>>()
+
     fun searchPictures(query: String) {
         viewModelScope.launch(IO) {
             repository.getSearchResultStream(query).apply {
                 unsplashSearchResponseLiveData.postValue(this)
             }
         }
     }
+
+    suspend fun insertPhoto(result: UnsplashSearchResponse.Result) {
```

Comment:
not the job of viewmodel to insert data in room it is repository job

Reply:
ok, I did it (remove them from viewmodel and make them private to repository) (I never use them in viewmodel)